src/playwright/test.ts

```diff
@@ -0,0 +1,348 @@
+          // advance what method will be called.
+          if (name in element && typeof element[name] === 'function') {
+            // @ts-expect-error See above.
+            //
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return
+            return element[name].call(element, arguments_);
+          } else {
+            throw new Error(
+              `${element.tagName} doesn't have a ${name} method.`,
+            );
+          }
+        },
+        { name, arguments_ },
+      );
+    });
+  },
+
+  /**
+   * @return An array of Rule IDs¹.
+   *
+   * 1: https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md
+   */
+  async isAccessible({ page }, use) {
+    await use(async (selector: string) => {
+      const result = await new AxeBuilder({ page }).include(selector).analyze();
+
+      return result.violations.map((violation) => {
+        return violation.id;
+      });
+    });
+  },
+
+  async mount({ page }, use) {
+    await use(async function mount(template: TemplateResult): Promise<void> {
+      for (const string of template.strings) {
+        const hasEventListener = /@\w+=$/.test(string);
+
+        if (hasEventListener) {
+          throw new Error(
+            "Event listeners aren't allowed in templates. They're far too complicated to handle relative to their usefulness. Use the `addEventListener()` fixture instead.",
+          );
+        }
+      }
+
+      // Stop the browser from attempting to fetch a favicon so there's not a 404 in the
+      // console.
+      await page.route('favicon.ico', (route) => {
+        route.abort();
+      });
+
+      await page.goto('/playwright.html');
+
+      const renderResult = render(template);
+      const html = await collectResult(renderResult);
+
+      const components = await page.evaluate((html) => {
+        document.body.innerHTML = html;
+
+        return [...document.body.querySelectorAll('*')]
+          .filter((element) => element.tagName.startsWith('GLIDE-CORE'))
+          .map((element) => element.tagName.toLowerCase());
+      }, html);
+
+      await Promise.all(
+        components.map((component) => {
+          let url: string | undefined;
+
+          // Digging through the manifest is only necessary because sub-component name don't
+          // map directly to filenames. For example, `glide-core-dropdown-option` maps to
+          // `dropdown.option.ts` on disk instead of mapping to `dropdown-option.ts`.
+          //
+          // If it mapped to the latter, we could simply set `url` to
+          // `src/${component.replace('glide-core-', '')}.ts` and call it a day.
+          for (const module of customElements.modules) {
+            const exportedComponent = module.exports.find(({ kind, name }) => {
+              return kind === 'custom-element-definition' && name === component;
```

Comment:
5. Now (back in Node.js land) iterate through every component found on the page and find it's path on disk via the manifest.